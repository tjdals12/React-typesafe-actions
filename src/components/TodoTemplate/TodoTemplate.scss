.page-template {
    width: 500px;

    margin-top: 5rem;
    margin-left: auto;
    margin-right: auto;
    padding: 2rem;

    background: white;
    box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.16);

    .title {
        text-align: center;
        font-family: 'Lobster', cursive;
        font-size: 4rem;
        font-weight: 600;
        margin: 0;
    }

    .content {
        margin-top: 2rem;
    }
}
