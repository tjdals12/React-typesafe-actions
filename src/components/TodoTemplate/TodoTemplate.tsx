import React from 'react';
import classNames from 'classnames/bind';
import styles from './TodoTemplate.scss';

const cx = classNames.bind(styles);

export default function PageTemplate({ children }: { children: React.ReactNode }): JSX.Element {
    return (
        <div className={cx('page-template')}>
            <h2 className={cx('title')}>TODO LIST</h2>
            <div className={cx('content')}>{children}</div>
        </div>
    );
}
