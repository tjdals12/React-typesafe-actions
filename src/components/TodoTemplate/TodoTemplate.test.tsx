import React from 'react';
import { render } from '@testing-library/react';
import TodoTemplate from './TodoTemplate';

describe('<TodoTemplate />', () => {
    it('matches snapshot', () => {
        const utils = render(<TodoTemplate />);
        expect(utils.container).toMatchSnapshot();
    });

    it('has a title and a content', () => {
        const utils = render(<TodoTemplate />);
        expect(utils.container.querySelector('.title')).not.toBeNull();
        expect(utils.container.querySelector('.content')).not.toBeNull();
    });
});
