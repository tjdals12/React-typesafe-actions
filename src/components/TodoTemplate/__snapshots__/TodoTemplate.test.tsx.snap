// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoTemplate /> matches snapshot 1`] = `
<div>
  <div
    class="page-template"
  >
    <h2
      class="title"
    >
      TODO LIST
    </h2>
    <div
      class="content"
    />
  </div>
</div>
`;
